File: Natom.Petshop.Gestion.Backend/Natom.Petshop.Gestion.Entities/DTO/FeatureFlagsDTO.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Natom.Petshop.Gestion.Entities.DTO
{
    public class FeatureFlagsDTO
    {
        [JsonProperty("stock")]
        public FeatureFlagsStockDTO Stock { get; set; }

        [JsonProperty("clientes")]
        public FeatureFlagsClientesDTO Clientes { get; set; }

        [JsonProperty("acceso")]
        public FeatureFlagsAccesoDTO Acceso { get; set; }

        [JsonProperty("pedidos")]
        public FeatureFlagsPedidosDTO Pedidos { get; set; }
    }

    public class FeatureFlagsPedidosDTO
    {
        [JsonProperty("fecha_y_hora_entrega_obligatorio")]
        public bool FechaYHoraEntregaObligatorio { get; set; }
    }

    public class FeatureFlagsClientesDTO
    {
        [JsonProperty("validar_solo_domicilio")]
        public bool ValidarSoloDomicilio { get; set; }
    }

    public class FeatureFlagsAccesoDTO
    {
        [JsonProperty("restringir_por_horario")]
        public bool RestringirPorHorario { get; set; }

        [JsonProperty("rango_horario_permitido_desde")]
        public int RangoHorarioPermitidoDesde { get; set; }

        [JsonProperty("rango_horario_permitido_hasta")]
        public int RangoHorarioPermitidoHasta { get; set; }
    }

    public class FeatureFlagsStockDTO
    {
        [JsonProperty("permitir_venta_con_stock_negativo")]
        public bool PermitirVentaConStockNegativo { get; set; }
    }
}

File: Natom.Petshop.Gestion.Frontend/src/app/classes/dto/_feature_flags/feature-flags.dto.ts
import { FeatureFlagsAccesoDTO } from "./feature-flags.acceso.dto";
import { FeatureFlagsClientesDTO } from "./feature-flags.clientes.dto";
import { FeatureFlagsPedidosDTO } from "./feature-flags.pedidos.dto";
import { FeatureFlagsStockDTO } from "./feature-flags.stock.dto";

export class FeatureFlagsDTO {
    public acceso: FeatureFlagsAccesoDTO;
    public clientes: FeatureFlagsClientesDTO;
    public stock: FeatureFlagsStockDTO;
    public pedidos: FeatureFlagsPedidosDTO;
}